### PR TITLE
Agent 278 improve test coverage for log processing.py

### PR DESCRIPTION
**Pull request by ArthurKamalov**
* add test for sampling_rules with rate<1.0
* add test for exclude files from config
* add test for reduction and sampling rules from config.

Removed 'if not _matches' in '__apply_redaction_rule',
because re.finditer result (iterator) can't be 'Falsy'.

### CODE REVIEW

File: scalyr_agent/tests/log_processing_test.py [hunk 1]
```diff
@@ -2212,6 +2251,76 @@ def test_find_matches_if_completely_finished(self):
...
+                                       })
+        config = self._create_log_config(self.__path_one, redaction_rules=[reduction_rule])
+        self.append_file(self.__path_one, "My line\n")
+        matcher = LogMatcher(self.__config, config)
+        matcher.finish()
+        processors = matcher.find_matches(dict(), dict(), copy_at_index_zero=True)
+
+        processor = processors[0]
+
+        s = processors[0].generate_status()
```
Comment (czerwingithub) [inline scalyr_agent/tests/log_processing_test.py]: Is this line needed?

File: scalyr_agent/log_processing.py [hunk 2]
```diff
@@ -2156,10 +2156,6 @@ def __replace_groups_with_hashed_content(re_ex, replacement_ex, line):
 
             _matches = re.finditer(re_ex, line)
 
-            if not _matches:
-                # if no matches, return the `line` as such
-                return line, None
-
```
[the remark this answers is not in the document]
Comment (ArthurKamalov) [reply scalyr_agent/log_processing.py]: If I am not mistaken, iterators can't be "Falsy"
```
>>bool(re.finditer(r'Hello', ""))
True
bool(list(re.finditer(r'Hello', "")))
False
```

File: scalyr_agent/tests/log_processing_test.py (hunk 1 above)
Comment (ArthurKamalov) [reply scalyr_agent/tests/log_processing_test.py]: Oops, bad cleanup, thank you.

File: scalyr_agent/log_processing.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (czerwingithub) [reply scalyr_agent/log_processing.py]: Ok, that makes sense.

**Pull request merged**